orangecontrib/spectroscopy/io/neaspec.py

```diff
@@ -310,4 +311,147 @@ def handle_data(self, data):
+        out_meta_floats = np.zeros((out_rows, 2), dtype="float") # Z, delta_X
+        out_meta_strs = np.zeros((out_rows, 1), dtype="<U10") # Channel
+
+        # getting the index of the columns that contain the relevant meta data
+        row_header_index = np.where(table_data_headers == "Row")[0][0]
+        col_header_index = np.where(table_data_headers == "Column")[0][0]
+        run_header_index = np.where(table_data_headers == "Run")[0][0]
+        z_header_index = np.where(table_data_headers == "Z")[0][0]
+        m_header_index = np.where(table_data_headers == "M")[0][0]
+
+        # filling the output data and metadata
+        # reading the data from the tableData rowise trough rows, columns and runs
+        # and then columnwise trough the channels and inserting it into
+        # the output table as rows using the data depth as the new data columns
+        channels_len = len(valid_channel_cols)
+        for row in range(rows):
+            for column in range(cols):
+                for run in range(runs):
+                    loop_step = row * cols * runs + column * runs + run
+                    input_table_index = depth * loop_step
+                    data_row = table_data_values[input_table_index, row_header_index]
+                    data_column = table_data_values[input_table_index, col_header_index]
+                    data_run = table_data_values[input_table_index, run_header_index]
+                    z_array = table_data_values[input_table_index : input_table_index + depth, z_header_index]
+                    m_array = table_data_values[input_table_index : input_table_index + depth, m_header_index]
+                    mean_z = np.mean(z_array)
+                    delta_m = np.mean(np.diff(m_array))
+                    delta_x = delta_m * 2 * 1e2 # converting interferometer distance to step size 
+                    for channel in range(channels_len):
+                        output_table_index = loop_step * channels_len + channel
+                        channel_data = table_data_values[
+                            input_table_index : input_table_index + depth,
+                            valid_channel_cols[channel],
+                        ]
+                        out_data[output_table_index, :] = channel_data.transpose()
+                        channel_name = table_data_headers[valid_channel_cols[channel]]
+                        # we are intentionally changing metadata here to match the expected output
+                        # by swapping the row and column index in the metadata
+                        out_meta_ints[output_table_index, :] = [data_row, data_column, data_run]
+                        out_meta_floats[output_table_index, :] = [mean_z, delta_x]
+                        out_meta_strs[output_table_index] = channel_name
+
+        # formatting the metadata as it is expected by Orange
+        metas = [
+            Orange.data.ContinuousVariable.make("column"),
```

Comment:
There is no need to use `.make` anymore. The reasons it exists are historical, but Orange keeps it for backward compatibility. Just deleting `.make` would achieve the same.